axelor-purchase/src/main/java/com/axelor/apps/purchase/script/ImportPurchaseOrder.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2018 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.axelor.apps.purchase.script;
+
+import java.util.Map;
+
+import com.axelor.apps.base.service.administration.SequenceService;
+import com.axelor.apps.purchase.db.PurchaseOrder;
+import com.axelor.apps.purchase.db.repo.PurchaseOrderRepository;
+import com.axelor.apps.purchase.service.PurchaseOrderService;
+import com.axelor.exception.AxelorException;
+import com.google.inject.Inject;
+
+public class ImportPurchaseOrder {
+	
+	@Inject
+	private SequenceService sequenceService;
+	
+	@Inject
+	private PurchaseOrderRepository purchaseOrderRepository;
+	
+	@Inject
+	private PurchaseOrderService purchaseOrderService;
+	
+	public Object importPurchaseOrder(Object bean, Map<String,Object> values) throws AxelorException {
+		assert bean instanceof PurchaseOrder;
+		
+		PurchaseOrder purchaseOrder = (PurchaseOrder) bean;
+		
+		if (purchaseOrder.getStatusSelect() == 1 || purchaseOrder.getStatusSelect() == 2) {
+			purchaseOrder.setPurchaseOrderSeq(sequenceService.getDraftSequenceNumber(purchaseOrder));
+		}
+		
+		purchaseOrder = purchaseOrderRepository.find(purchaseOrder.getId());
```

Comment:
Why you fetched managed instance of PO, and is there meaning of setting sequence in previous line if you  fetch on next line ?